crypto/objects/o_names.c

```diff
@@ -157,11 +174,13 @@ const char *OBJ_NAME_get(const char *name, int type)
 {
     OBJ_NAME on, *ret;
     int num = 0, alias;
+    const char *value = NULL;
 
     if (name == NULL)
         return (NULL);
-    if ((names_lh == NULL) && !OBJ_NAME_init())
+    if (!OBJ_NAME_init())
         return (NULL);
+    CRYPTO_THREAD_read_lock(lock);
```

Comment:
Yes, that is a great idea. We have CRYPTO_atomic_add() for that. Do you want to pick that up as part of this PR, or should we defer it to another one?